src/test/java/emissary/test/core/junit5/ExtractionTest.java

```diff
@@ -368,19 +368,54 @@ protected void checkAnswers(Document answers, IBaseDataObject payload, List<IBas
     protected void checkAnswers(Element el, IBaseDataObject payload, @Nullable List<IBaseDataObject> attachments, String tname)
             throws DataConversionException {
 
-        int numAtt = JDOMUtil.getChildIntValue(el, "numAttachments");
```

Comment:
I believe we might be overcomplicating this slightly. The logic that verifies the attachment number against the provided attachments could remain as is. Our main adjustment would involve altering how we extract answers from the XML. Maybe if we did something like:
```
        int numAtt = -1;
        long numAttElements = -1;
        List<Element> numAttachments = el.getChildren("numAttachments");
        for (Element numAttEl : numAttachments) {
            if (verifyOs(numAttEl)) {
                numAtt = Integer.parseInt(numAttEl.getValue());
                numAttElements = el.getChildren().stream().filter(c -> c.getName().startsWith(ATTACHMENT_ELEMENT_PREFIX) && verifyOs(c)).count();
                break;
            }
        }
```
Thoughts?